kafka-client/src/main/java/dev/responsive/kafka/store/CommitBuffer.java

```diff
@@ -155,22 +155,40 @@ static <K> CommitBuffer<K> from(
     this.subPartitioner = subPartitioner;
     this.clock = clock;
     this.maxConcurrentWrites = maxConcurrentWrites;
-    lastFlush = clock.get();
+
+    final String logPrefix = String.format("commit-buffer-%s[%d] ", tableName, partition);
+    final LogContext logContext = new LogContext(logPrefix);
+    log = logContext.logger(CommitBuffer.class);
   }
 
   public void init() {
-    subPartitioner.all(partition).forEach(p -> client.initializeOffset(tableName, p));
-
-    // when we initialize a new state store, we should revoke all ongoing
-    // transactions to the remote store - otherwise it is possible that
-    // there's a hanging transaction that will fence this writer if there
-    // is no data to restore
-    final long offset = offset();
-    if (offset >= 0) {
-      subPartitioner
-          .all(partition)
-          .forEach(p -> client.execute(client.revokePermit(tableName, p, offset)));
+    subPartitioner.all(partition).forEach(p -> client.initializeMetadata(tableName, p));
+    lastFlush = clock.get();
+
+    // attempt to reserve an epoch - all epoch reservations will be done
+    // under the first sub-partition and then "broadcast" to the other
+    // partitions
+    final int basePartition = subPartitioner.first(partition);
+    this.epoch = client.metadata(tableName, basePartition).epoch + 1;
+
+    for (final int subPartition : subPartitioner.all(partition).toArray()) {
```

Comment:
do we have a test that ensures these are iterated in order?

Reply:
will add 👍